Cubical/Algebra/CommAlgebra/FreeCommAlgebra/Properties.agda

```diff
@@ -46,10 +48,110 @@ module Theory {R : CommRing ℓ} {I : Type ℓ'} where
          using (0r; 1r)
          renaming (_·_ to _·r_; _+_ to _+r_; ·Comm to ·r-comm; ·IdR to ·r-rid)
 
+  module C = Construction
+  open C using (var; const)
+
+  {-
+    Construction of the elimProp eliminator.
+  -}
+  module _
+    {P : ⟨ R [ I ] ⟩ → Type ℓ''}
+    (isPropP : {x : _} → isProp (P x))
+    (onVar : {x : I} → P (var x))
+    (onConst : {r : ⟨ R ⟩} → P (const r))
+    (on+ : {x y : ⟨ R [ I ] ⟩} → P x → P y → P (x C.+ y))
+    (on· : {x y : ⟨ R [ I ] ⟩} → P x → P y → P (x C.· y))
+    where
+
+    private
+      on- : {x : ⟨ R [ I ] ⟩} → P x → P (C.- x)
+      on- {x} Px = subst P (minusByMult x) (on· onConst Px)
+        where open ModuleTheory _ (Algebra→Module (CommAlgebra→Algebra (R [ I ])))
+
+      -- A helper to deal with the path constructor cases.
+      mkPathP :
+        {x0 x1 : ⟨ R [ I ] ⟩} →
+        (p : x0 ≡ x1) →
+        (Px0 : P (x0)) →
+        (Px1 : P (x1)) →
+        PathP (λ i → P (p i)) Px0 Px1
+      mkPathP _ = isProp→PathP λ i → isPropP
+
+    elimProp : ((x : _) → P x)
+
+    elimProp (var _) = onVar
+    elimProp (const _) = onConst
+    elimProp (x C.+ y) = on+ (elimProp x) (elimProp y)
+    elimProp (C.- x) = on- (elimProp x)
+    elimProp (x C.· y) = on· (elimProp x) (elimProp y)
+
+    elimProp (C.+-assoc x y z i) =
+      mkPathP (C.+-assoc x y z)
+        (on+ (elimProp x) (on+ (elimProp y) (elimProp z)))
+        (on+ (on+ (elimProp x) (elimProp y)) (elimProp z))
+        i
+    elimProp (C.+-rid x i) =
+      mkPathP (C.+-rid x)
+        (on+ (elimProp x) onConst)
+        (elimProp x)
+        i
+    elimProp (C.+-rinv x i) =
+      mkPathP (C.+-rinv x)
+        (on+ (elimProp x) (on- (elimProp x)))
+        onConst
+        i
+    elimProp (C.+-comm x y i) =
+      mkPathP (C.+-comm x y)
+        (on+ (elimProp x) (elimProp y))
+        (on+ (elimProp y) (elimProp x))
+        i
+    elimProp (C.·-assoc x y z i) =
```

Comment:
I would be interested if all these path constructor cases can be made any shorter than this.

Reply:
No good ideas here. Did you try dropping the 'i' on both sides?